src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt

```diff
@@ -18,7 +23,9 @@ interface ApplicationRepository : JpaRepository<ApplicationEntity, UUID> {
 
```

Comment:
If there's a situation where a user can move from CAS1 to CAS3 (or vice versa), it might be worth updating the `ApplicationRepository.findAllByCreatedByUser_Id` method to also filter by the application entity class to avoid applications from the wrong service from showing up:

```kotlin
@Query("SELECT a FROM ApplicationEntity a WHERE TYPE(a) = :type AND a.createdByUser.id == :id")
fun <T : ApplicationEntity> findAllByCreatedByUser_Id(id: UUID, type: Class<T>): List<ApplicationEntity>
```

Reply:
Good spot, have added in 👍